alfresco-solrapi-client/alfresco-solrapi-client-ditto/src/integration-test/java/eu/xenit/alfresco/solrapi/client/ditto/DittoClientNoLiveNodeExistsReproduction.java

```diff
@@ -0,0 +1,85 @@
+package eu.xenit.alfresco.solrapi.client.ditto;
+
+import eu.xenit.alfresco.solrapi.client.spi.SolrApiClient;
+import eu.xenit.alfresco.solrapi.client.tests.GetMetadataNoLiveNodeExistsReproduction;
+import eu.xenit.testing.ditto.api.AlfrescoDataSet;
+import eu.xenit.testing.ditto.api.DataSetBuilder;
+import eu.xenit.testing.ditto.api.data.ContentModel.Content;
+import eu.xenit.testing.ditto.api.data.ContentModel.System;
+import eu.xenit.testing.ditto.api.data.ContentModel.Version2;
+import eu.xenit.testing.ditto.api.model.Node;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class DittoClientNoLiveNodeExistsReproduction implements GetMetadataNoLiveNodeExistsReproduction {
+
+    private AlfrescoDataSet dataSet;
+
+    @Override
+    public SolrApiClient solrApiClient() {
+        return new SolrApiFakeClient(dataSet);
+    }
+
+    @Override
+    public String createVersionableNode(String name) {
+        final String uuid = UUID.randomUUID().toString();
+
+        dataSet = AlfrescoDataSet.builder()
+                .addTransaction(txn -> {
+                    txn.addNode(node -> {
+                        node.storeRefProtocol("workspace");
+                        node.storeRefIdentifier("version2Store");
+                        node.name(name);
+                        node.type(Content.CONTENT);
+                        node.property(Version2.FROZEN_NODE_REF, "workspace://SpacesStore/" + uuid);
+                    });
+                    txn.addNode(node -> {
+                        node.uuid(uuid);
+                        node.name(name);
+                        node.type(Content.CONTENT);
+                    });
+                })
+                .build();
+
+        return uuid;
+    }
+
+    @Override
+    public List<Long> getVersionNodeIds(String liveNodeUuid) {
+        return dataSet.getNodeView().stream()
+                .filter(node -> ("workspace://SpacesStore/" + liveNodeUuid).equals(
+                        node.getProperties().get(Version2.FROZEN_NODE_REF).orElse(null)))
+                .map(Node::getNodeId)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void deleteNode(String liveNodeUuid) {
+        DataSetBuilder dataSetBuilder = AlfrescoDataSet.builder();
```

Comment:
You faked this here, because you are waiting for ditto to support editing stuff ?

Reply:
Yeah so I introduced a minimal scenario to reproduce this issue:

1.  Create a node with the cm:versionable aspect. 
    * In a "real" Alfresco this implicates an initial `workspace://version2Store` version is automatically created
    * In Ditto I mimicked this behavior by creating an `AlfrescoDataSet` containing both the `ws://SpacesStore` and `ws://version2Store` nodes.

2. Delete the live node
   * Ditto does not yet support the deletion of nodes or the manipulation of the `AlfrescoDataSet`. Therefore I had to "hack" my way around this by recreating the `AlfrescoDataSet`. Indeed, if Ditto is refactored to support manipulation of the `AlfrescoDataSet` this can be simplified.